Compilation issue of Kokkos-Kernels on HIP backend

We encountered a configuration failure of Kokkos-kernels against current develop.
Here is the minimal reproducer.
I am using `rocm 6.3.3`.
It may be related to #8747

```bash
export Kokkos_ROOT=<your-path-to-kokkos>

cmake -D CMAKE_CXX_STANDARD=20 \
            -DCMAKE_CXX_COMPILER=hipcc \
            -DCMAKE_INSTALL_PREFIX=${Kokkos_ROOT} \
            -D Kokkos_ENABLE_DEPRECATED_CODE_4=OFF \
            -D Kokkos_ENABLE_DEPRECATION_WARNINGS=OFF \
            -D Kokkos_ENABLE_SERIAL=ON \
            -DKokkos_ENABLE_HIP=ON \
            -DKokkos_ENABLE_HIP_RELOCATABLE_DEVICE_CODE=ON \
            -DKokkos_ENABLE_ROCTHRUST=OFF \
            -DKokkos_ARCH_AMD_GFX90A=ON \
            -B build

cmake --build build 
cmake --install build

export KokkosKernels_ROOT=<your-path-to-kokkos-kernels>

cmake -DCMAKE_CXX_STANDARD=20 \
            -DCMAKE_CXX_COMPILER=hipcc \
            -DCMAKE_INSTALL_PREFIX=${KokkosKernels_ROOT} \
            -DCMAKE_PREFIX_PATH=${Kokkos_ROOT} \
            -DKokkosKernels_ADD_DEFAULT_ETI=OFF \
            -DKokkosKernels_ENABLE_ALL_COMPONENTS=OFF \
            -DKokkosKernels_ENABLE_COMPONENT_BLAS=ON \
            -DKokkosKernels_ENABLE_COMPONENT_BATCHED=ON \
            -DKokkosKernels_ENABLE_COMPONENT_LAPACK=OFF \
            -DKokkosKernels_ENABLE_TPL_BLAS=OFF \
            -DKokkosKernels_ENABLE_TPL_CUSOLVER=OFF \
            -DKokkosKernels_ENABLE_TPL_LAPACK=OFF \
            -B build
            
cmake --build build 
cmake --install build
```

I got the following errors

```
-- The CXX compiler identification is Clang 18.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/rocm-6.3.3/bin/hipcc - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The project name is: KokkosKernels
-- Enabled Kokkos devices: SERIAL;HIP
-- Found Kokkos version 5.0.99 at /lus/home/CT6/cin4492/yasahi/scratch/lib/kokkos/lib64/cmake/Kokkos
CMake Warning at cmake/kokkoskernels_features.cmake:16 (message):
  Disabling SUPERNODAL_SPTRSV - this capability is only supported with
  LayoutLeft
Call Stack (most recent call first):
  CMakeLists.txt:218 (include)



================================
Kokkos Kernels version: 5.0.99
================================
Kokkos Kernels ETI Types
   Devices:  <HIP,HIPSpace>
   Scalars:  
   Ordinals: 
   Offsets:  
   Layouts:  

Kokkos Kernels components
   COMMON:    ON
   BATCHED:   ON
   BLAS:      ON
   LAPACK:    OFF
   GRAPH:     OFF
   SPARSE:    OFF
   ODE:       OFF

Kokkos Kernels TPLs
   (None)
================================

-- Creating ETI files for Batched_Gemm_nt_nt_bll
-- Creating ETI files for Batched_Gemm_nt_nt_blr
-- Creating ETI files for Batched_Gemm_nt_t_bll
-- Creating ETI files for Batched_Gemm_nt_t_blr
-- Creating ETI files for Batched_Gemm_t_nt_bll
-- Creating ETI files for Batched_Gemm_t_nt_blr
-- Creating ETI files for Batched_Gemm_t_t_bll
-- Creating ETI files for Batched_Gemm_t_t_blr
-- Creating ETI files for Blas1_abs
-- Creating ETI files for Blas1_abs_mv
-- Creating ETI files for Blas1_scal
-- Creating ETI files for Blas1_scal_mv
-- Creating ETI files for Blas1_dot
-- Creating ETI files for Blas1_dot_mv
-- Creating ETI files for Blas1_axpby
-- Creating ETI files for Blas1_axpby_mv
-- Creating ETI files for Blas1_update
-- Creating ETI files for Blas1_update_mv
-- Creating ETI files for Blas1_sum
-- Creating ETI files for Blas1_sum_mv
-- Creating ETI files for Blas1_nrm1
-- Creating ETI files for Blas1_nrm1_mv
-- Creating ETI files for Blas1_nrm2w
-- Creating ETI files for Blas1_nrm2w_mv
-- Creating ETI files for Blas1_nrminf
-- Creating ETI files for Blas1_nrminf_mv
-- Creating ETI files for Blas1_iamax
-- Creating ETI files for Blas1_iamax_mv
-- Creating ETI files for Blas1_nrm2
-- Creating ETI files for Blas1_nrm2_mv
-- Creating ETI files for Blas1_mult
-- Creating ETI files for Blas1_mult_mv
-- Creating ETI files for Blas1_reciprocal
-- Creating ETI files for Blas1_reciprocal_mv
-- Creating ETI files for Blas1_rot
-- Creating ETI files for Blas1_rotg
-- Creating ETI files for Blas1_swap
-- Creating ETI files for Blas2_gemv
-- Creating ETI files for Blas2_ger
-- Creating ETI files for Blas2_syr
-- Creating ETI files for Blas2_syr2
-- Creating ETI files for Blas3_gemm
-- Creating ETI files for Blas3_trsm
-- Creating ETI files for Blas3_trmm
-- Creating ETI files for Blas1_rotm
-- Creating ETI files for Blas1_rotmg
-- Configuring done (4.5s)
CMake Error at /lus/home/CT6/cin4492/yasahi/scratch/lib/kokkos/lib64/cmake/Kokkos/KokkosTargets.cmake:61 (set_target_properties):
  The link interface of target "Kokkos::ROCM" contains:

    hip::device

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /lus/home/CT6/cin4492/yasahi/scratch/lib/kokkos/lib64/cmake/Kokkos/KokkosConfig.cmake:42 (include)
  CMakeLists.txt:120 (find_package)


-- Generating done (0.2s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

--- a/cmake/Modules/FindTPLROCM.cmake
+++ b/cmake/Modules/FindTPLROCM.cmake
@@ -1,3 +1,4 @@
 find_package(hip REQUIRED PATHS ${ROCM_PATH} $ENV{ROCM_PATH})
 
 kokkos_create_imported_tpl(ROCM INTERFACE LINK_LIBRARIES hip::device)
+kokkos_export_cmake_tpl(hip REQUIRED)

